Fixes links within mdx h2/h3, adjusts `code` css [Fixes #1843]

wackerow opened the pull request:
## Description
- This adds a css selector to the `Header2` and `Header3` components within `SharedStyledComponents.js`, selecting for `<code>` children, and adjusts height to be relative to its parent.
- Changed the `a` sub-styling on these two components to `a.header-anchor` to select _only_ for the link being generator when parsing mdx (the chain-link to the left of the header). This leaves intentionally placed links within a header to render normally.
- Note this does not actually create nested anchor tags (as I previously expressed concern for), because the actual header text is not a link (only the side chain-link icon), and the ToC only strips the text, and does not apply a second link or additional styling (like `code` styling for example)

## Related Issue #1843

![image](https://user-images.githubusercontent.com/54227730/99916168-5ddae600-2cbd-11eb-85d4-8552b744766d.png)

wackerow commented:
![image](https://user-images.githubusercontent.com/54227730/99918101-954f8f80-2cc9-11eb-9472-fdaa3c2d9689.png)
![image](https://user-images.githubusercontent.com/54227730/99918109-9ed8f780-2cc9-11eb-8763-1c22aa46346b.png)

Noticed a very slight alignment issue when rendering these `<code>` snippets, which seems to render some of them perfectly in line, and others nudged down (and seems to happen at any screen size).

Let me know if anyone has thoughts on why this is rendering differently on two different places on the screen at the same time.

@ryancreatescopy Btw, I chose the `em` units to be a little less than one because using 1 made this text look almost bolded and the characters became slightly taller than surrounding text (different font-family I assume)... Let me know what you think, or if you have a different thought on this entirely.

ryancreatescopy commented:
To me, `1em` looks better. It doesn't seem to bolden it at all. I think removing the padding from it might be useful too.
![image](https://user-images.githubusercontent.com/40891631/99956539-19634f00-2d7e-11eb-8d24-aa72d3a888aa.png)


When it's in the middle of a sentence though, we get this weird space after the code which goes up to the following word. Do you think we should remove that?

wackerow commented:
@ryancreatescopy Ah... yeah ideally, hah... did most in dark mode which is a little harder to see the background.. I'll take a look, and also switch back to 1em

wackerow commented:
Previous commit adjusted this styling within "SharedStyledComponents", targeting H2/H3... but after reviewing, it seems more appropriate to blanket this across all code snippets from the `layout.css` file, since all code snippets are currently forced to `0.85rem`, despite surrounding context, and gap glitch following code snippets occurs everywhere, even if outside a header.

- Reverted `SharedStyledComponents` changes that related to the code snippets, and placed this logic in `layout.css` by adjusting `font-size: 0.85rem` to `font-size: 1em`. This will allow code snippets to match the font size of its surrounding text, instead of referencing the 'root em'.
- Trimmed vertical padding on code snippets by 25% (now `0.15em`)
- Removed `:after` styling to remove gap being placed after code snippets

### Screenshots as of a47177a:
![image](https://user-images.githubusercontent.com/54227730/100032044-ee8fef80-2dab-11eb-9789-c585b2d3fb2a.png)
![image](https://user-images.githubusercontent.com/54227730/100032024-dfa93d00-2dab-11eb-973f-b50b2802eae9.png)

ryancreatescopy commented:
This looks good to me – looks much better in the UI! I defer to @samajammin on the technical implementation.

samajammin reviewed (commented):
LGTM! Nice work!
